### PR TITLE
Fix list switch and card selection 

### LINKED ISSUE #203: Selecting the same index does not update browser panel
**Describe the bug**
If the same index is selected on 2 different Lists, swapping between the panels does not update the browser panel.

**To Reproduce**
1. Have 2 populated lists
2. Select index 1 from list 1 (sr 1)
3. Select index 1 from list 2 (si 1)
4. Select index 1 from list 1 (sr 1)
5. Select index 1 from list 2 (si 1)
6. Observe that the browser panel is not updating properly. 

**Expected behavior**
Browser panel should update when swapping between different Lists.

### LINKED ISSUE #212: Move files into respective packages
Moved all `ui` events into respective feature's packages (`menu`, `ingredient`, etc...).

### PR DESCRIPTION
**Pull request by AZhiKai**
Fixes #203 
Resolves #212 

**Initial findings**
It has been established that the bug is caused by the following chunk of code

```java
 private void setEventHandlerForSelectionChangeEvent() {
        reservationListView.getSelectionModel().selectedItemProperty()
                .addListener((observable, oldValue, newValue) -> {
                    if (newValue != null) {
                        logger.fine("Selection in reservation list panel changed to : '" + newValue + "'");
                        raise(new ReservationPanelSelectionChangedEvent(newValue));
                    }
                });
    }
```
If one examines the code closely, the event will only be raised if and only if the `oldValue` and `newValue` is different. Therefore, by re-selecting the same card, it doesn't fire the event. 

It is also worthy to point out that we shouldn't be confused at the `scrollTo` raising the event as well. If you also notice, `scrollTo` is triggered by the `JumpTo...ListEvent` which will only happen if we select the card using commands (e.g. `sr 1`, `si 1`). In this case, we are firing the event as long as either a click OR command happen.

In addition, a rogue implementation in the previous few commits resulted in severe memory leak if the application is used for a prolong period of time without shutting down. So this PR fixes that memory leak as well! 😄 

**Resolution**
To resolve this bug, we simply get rid of the entire method, and implement a mouse event handler which is guaranteed to always fire as long as you click on the card. 

**READY FOR REVIEW! COVERAGE TO BE HANDLED BY THE RESPECTIVE CONTRIBUTOR**

### CODE REVIEW

**Comment by AZhiKai** [comment]
@yican95  and @m4dkip  I'll leave the test cases to you both. 😃

File: src/main/resources/view/default.html [hunk 1]
```diff
@@ -8,7 +8,7 @@
             margin: 0;
         }
 
-/*        .background {
+        /*.background {
```
Comment (m4dkip) [inline src/main/resources/view/default.html]: Is it safe to re-enable the splash screen?

Comment (AZhiKai) [reply src/main/resources/view/default.html]: I supposed so. Let me test it real quick. :)

Comment (AZhiKai) [reply src/main/resources/view/default.html]: @m4dkip  Yup it works now. 😄

**Pull request merged**